### PR TITLE
Rename `TieredMulti` methods to be more descriptive

### LINKED ISSUE #1341: Rename methods in `TieredMulti` to be more descriptive
For context I pasted the generic `T` implemenation for `TieredMulti` below. 
I put the new suggested names above the method in a comment.

Probably take a look at `Tiered` as well if these renames happen.

thoughts? 
@dpc 

Short: 
```rust
impl<T> TieredMulti<T> {
    pub fn new(map: BTreeMap<Amount, Vec<T>>) -> Self { TieredMulti(map) }

    // good. or notes_amount() or notes_value() or total_notes_value()
    pub fn total_amount(&self) -> Amount { }

    //  count_notes()
    pub fn item_count(&self) -> usize { }

    // count_tiers()
    pub fn tier_count(&self) -> usize { }

    // iter_tiers() or keys()
    pub fn tiers(&self) -> impl Iterator<Item = &Amount> { }

    pub fn is_empty(&self) -> bool { }

    pub fn map<F, N, E>(self, f: F) -> Result<TieredMulti<N>, E> }

    pub fn structural_eq<O>(&self, other: &TieredMulti<O>) -> bool { }

    // iter() ,it also just calls iter()
    pub fn iter_tiers(&self) -> impl Iterator<Item = (&Amount, &Vec<T>)> { }

    // Note: order of the elements is important here: from lowest tiers to highest, then in order of elements in the Vec
    pub fn iter_items(&self) -> impl Iterator<Item = (Amount, &T)> + DoubleEndedIterator { }

    // Note: order of the elements is important here: from lowest tiers to highest, then in order of elements in the Vec
    pub fn into_iter_items(self) -> impl Iterator<Item = (Amount, T)> + DoubleEndedIterator { }

    /// Returns the max number of notes for any given denomination tier
    // max_number_of_notes(), would be longer but descriptive. I'd also change the documentation since there is no "given denomination tier"
    pub fn max_tier_len(&self) -> usize { }

    /// Verifies that all `self.keys()` are contained in the keys of the given parameter `Tiered`
    // tiers_exist_in()
    pub fn check_tiers<K>(&self, keys: &Tiered<K>) -> Result<(), InvalidAmountTierError> { }

    pub fn get(&self, amt: Amount) -> Option<&Vec<T>> { }

    // TODO: Get rid of it. It might be used to break useful invariants (like making sure there are no empty `Vec`s after removal)
    pub fn get_mut(&mut self, amt: Amount) -> Option<&mut Vec<T>> { }
}
```

<details>
<summary>with code implemenation </summary>

```rust
impl<T> TieredMulti<T> {
    pub fn new(map: BTreeMap<Amount, Vec<T>>) -> Self {
        TieredMulti(map)
    }

    // good. or notes_amount() or notes_value() or total_notes_value()
    pub fn total_amount(&self) -> Amount {
        let milli_sat = self
            .0
            .iter()
            .map(|(tier, coins)| tier.msats * (coins.len() as u64))
            .sum();
        Amount { msats: milli_sat }
    }

    //  why not count_notes()
    pub fn item_count(&self) -> usize {
        self.0.values().map(|coins| coins.len()).sum()
    }

    // count_tiers()
    pub fn tier_count(&self) -> usize {
        self.0.len()
    }

    // iter_tiers() or keys()
    pub fn tiers(&self) -> impl Iterator<Item = &Amount> {
        self.0.keys()
    }

    pub fn is_empty(&self) -> bool {
        self.item_count() == 0
    }

    pub fn map<F, N, E>(self, f: F) -> Result<TieredMulti<N>, E>
    where
        F: Fn(Amount, T) -> Result<N, E>,
    {
        let res = self
            .0
            .into_iter()
            .map(|(amt, coins)| -> Result<_, E> {
                let coins = coins
                    .into_iter()
                    .map(|coin| f(amt, coin))
                    .collect::<Result<Vec<_>, E>>()?;
                Ok((amt, coins))
            })
            .collect::<Result<BTreeMap<Amount, Vec<N>>, E>>()?;

        Ok(TieredMulti(res))
    }

    pub fn structural_eq<O>(&self, other: &TieredMulti<O>) -> bool {
        let tier_eq = self.0.keys().eq(other.0.keys());
        let per_tier_eq = self
            .0
            .values()
            .zip(other.0.values())
            .all(|(c1, c2)| c1.len() == c2.len());

        tier_eq && per_tier_eq
    }

    // iter() ,it also just calls iter()
    pub fn iter_tiers(&self) -> impl Iterator<Item = (&Amount, &Vec<T>)> {
        self.0.iter()
    }

    // Note: order of the elements is important here: from lowest tiers to highest, then in order of elements in the Vec
    pub fn iter_items(&self) -> impl Iterator<Item = (Amount, &T)> + DoubleEndedIterator {
        self.0
            .iter()
            .flat_map(|(amt, coins)| coins.iter().map(move |c| (*amt, c)))
    }
    // Note: order of the elements is important here: from lowest tiers to highest, then in order of elements in the Vec
    pub fn into_iter_items(self) -> impl Iterator<Item = (Amount, T)> + DoubleEndedIterator {
        self.0
            .into_iter()
            .flat_map(|(amt, coins)| coins.into_iter().map(move |c| (amt, c)))
    }

    /// Returns the max number of notes for any given denomination tier
    // max_number_of_notes(), would be longer but descriptive
    pub fn max_tier_len(&self) -> usize {
        self.0.values().map(|notes| notes.len()).max().unwrap_or(0)
    }

    /// Verifies that all `self.keys()` are contained in the keys of the given parameter `Tiered`
    // tiers_exist_in()
    pub fn check_tiers<K>(&self, keys: &Tiered<K>) -> Result<(), InvalidAmountTierError> {
        match self.0.keys().find(|&amt| keys.get(*amt).is_none()) {
            Some(amt) => Err(InvalidAmountTierError(*amt)),
            None => Ok(()),
        }
    }

    pub fn get(&self, amt: Amount) -> Option<&Vec<T>> {
        self.0.get(&amt)
    }

    // TODO: Get rid of it. It might be used to break useful invariants (like making sure there are no empty `Vec`s after removal)
    pub fn get_mut(&mut self, amt: Amount) -> Option<&mut Vec<T>> {
        self.0.get_mut(&amt)
}
```
</details>

### PR DESCRIPTION
**Pull request by mxz42**
closes #1341

### CODE REVIEW

File: fedimint-api/src/tiered_multi.rs [hunk 1]
```diff
@@ -78,35 +86,49 @@ impl<T> TieredMulti<T> {
...
+        // elements stays consistent.
         self.0
             .into_iter()
             .flat_map(|(amt, coins)| coins.into_iter().map(move |c| (amt, c)))
     }
 
-    /// Returns the max number of notes for any given denomination tier
-    pub fn max_tier_len(&self) -> usize {
+    /// Returns the maximum length of all tier vectors
+    pub fn max_value_len(&self) -> usize {
```
Comment (dpc) [inline fedimint-api/src/tiered_multi.rs:119]: Now that I look at it: comment doesn't fit the name well.

`longest_tier_len` maybe?

File: fedimint-api/src/tiered_multi.rs [hunk 2]
```diff
@@ -78,35 +86,49 @@ impl<T> TieredMulti<T> {
...
         self.0.values().map(|notes| notes.len()).max().unwrap_or(0)
     }
 
-    pub fn check_tiers<K>(&self, keys: &Tiered<K>) -> Result<(), InvalidAmountTierError> {
+    /// Verifies that all keys in `self` are present in the keys of the given parameter `Tiered`
+    pub fn tiers_exist_in<K>(&self, keys: &Tiered<K>) -> Result<(), InvalidAmountTierError> {
```
Comment (dpc) [inline fedimint-api/src/tiered_multi.rs:124]: Similar here.  `all_tier_exist_in` maybe?

**Comment by dpc** [comment]
@mxz42 Now that I've seen the comments, I think these two could be named different. Otherwise LGTM.

File: fedimint-api/src/tiered_multi.rs (hunk 1 above)
Comment (mxz42) [reply fedimint-api/src/tiered_multi.rs:119]: yeah, not the best name. how about `max_len()` or `max_vec_len()` 

I'll rewrite the comment, too. something like: `Returns the length of the longest vector of all tiers`

Comment (dpc) [reply fedimint-api/src/tiered_multi.rs:119]: Seems to me like `longest` will have to be there in the name. `max` is kind of vague - is it a lenght of "max" tier (highest amount)?

File: fedimint-api/src/tiered_multi.rs (hunk 2 above)
Comment (mxz42) [reply fedimint-api/src/tiered_multi.rs:124]: mh, I think the `all_tier` in `all_tier_exist_in` is represented with `tiers` in `tiers_exist_in`.  Or `all_tiers_exist_in` (note the `s`)

Comment (dpc) [reply fedimint-api/src/tiered_multi.rs:124]: Specifically it's a `any` vs `all` clarification.

Comment (mxz42) [reply fedimint-api/src/tiered_multi.rs:124]: let's go for `all_tiers_exist_in` then

File: fedimint-api/src/tiered_multi.rs (hunk 1 above)
Comment (mxz42) [reply fedimint-api/src/tiered_multi.rs:119]: mh, ok. on second thought, I like `longest_tier_len` -> give me the len of the longest tier :+1:

**Review by dpc — APPROVED** [review]

**Comment by dpc** [comment]
LGTM, possibly @elsirion wants to take a look before we land.

**Review by elsirion — APPROVED** [review]

**Pull request merged**